sdk/src/Services/S3/Custom/Util/AmazonS3Uri.cs

```diff
@@ -188,16 +188,19 @@ public static bool TryParseAmazonS3Uri(string uri, out AmazonS3Uri amazonS3Uri)
         /// <returns>true if the Uri is an AmazonS3Endpoint, and the out paramter has been filled in, false otherwise</returns>
         public static bool TryParseAmazonS3Uri(Uri uri, out AmazonS3Uri amazonS3Uri)
         {
-            if (IsAmazonS3Endpoint(uri))
-            {
-                amazonS3Uri = new AmazonS3Uri(uri);
-                return true;
-            }
-            else
+            try
             {
-                amazonS3Uri = null;
-                return false;
+                if (IsAmazonS3Endpoint(uri))
+                {
+                    amazonS3Uri = new AmazonS3Uri(uri);
+                    return true;
+                }
             }
+            // intentionally suppress all exceptions, because this is a try method
+            catch { }
```

Comment:
@ppittle - This a causes a Warning CA1031, which in VS2019 defaults to treating as an error:
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1031?view=vs-2019

Reply:
suppressed